app/src/main/java/it/quip/android/activity/CreateQuipActivity.java

```diff
@@ -62,24 +69,32 @@ private void showShareFragment() {
 
     @Override
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (requestCode == SHARE_QUIP_REQUEST) {
-            List<Circle> selectedCircles = ShareQuipActivity.getCircles();
-            if (null != selectedCircles) {
-                createQuip(selectedCircles);
-            }
-
-            Intent i = new Intent();
-            if (selectedCircles.size() > 0) {
-                i.putExtra(CREATED_QUIP_CIRCLE_ID, selectedCircles.get(0).getObjectId());
+        if (resultCode == RESULT_OK) {
+            if (requestCode == SHARE_QUIP_REQUEST) {
+                List<Circle> selectedCircles = ShareQuipActivity.getCircles();
```

Comment:
Would be nice to keep this to just the branching logic (checking request/result codes) and then hiding away the action in a private method